Packs/CheckPoint_FW/Integrations/CheckPoint_FW/CheckPoint_FW.py

```diff
@@ -0,0 +1,833 @@
+        testing(bool): is this function call is for testing purposes.
+    Returns:
+        str: the markdown to display inside Demisto.
+        dict: the context to return into Demisto.
+        dict: the report from CheckPoint (used for debugging).
+    """
+    common_output = {
+        'name': result.get('name'),
+        'uid': result.get('uid'),
+        'type': result.get('type'),
+        'domain_name': result.get('domain').get('name'),
+        'domain_uid': result.get('domain').get('uid'),
+        'domain_type': result.get('domain').get('type'),
+        'meta_info:creator': result.get('meta-info').get('creator'),
+        'meta_info:last_modifier': result.get('meta-info').get('last-modifier')
+    }
+    unique_outputs = {}
+    if endpoint == 'application-site':
+        unique_outputs = {
+            'application-id': result.get('application-id'),
+            'description': result.get('description'),
+            'url-list': result.get('url-list'),
+        }
+    elif endpoint == 'access-rule':
+        unique_outputs = {
+            'layer': result.get('layer'),
+            'enabled': result.get('enabled'),
+        }
+    elif endpoint == 'address range':
+        unique_outputs = {
+            'ipv4-address-first': result.get('ipv4-address-first'),
+            'ipv4-address-last': result.get('ipv4-address-last'),
+            'ipv6-address-first': result.get('ipv6-address-first'),
+            'ipv6-address-last': result.get('ipv6-address-last'),
+            'read-only': result.get('read-only')
+        }
+    elif endpoint == 'group':
+        if result.get('groups'):
+            unique_outputs = {
+                'groups-name': result.get('groups')[0]
+            }
+    elif endpoint == 'host':
+        unique_outputs = {
+            'ipv4-address': result.get('ipv4-address'),
+            'ipv6-address': result.get('ipv4-address'),
+            'read-only': result.get('read-only')
+        }
+    common_output.update(unique_outputs)
+    outputs = {f'Checkpoint.{endpoint}': common_output}
+
+    table_data = outputs[f'Checkpoint.{endpoint}']
+    readable_output = tableToMarkdown(f'CheckPoint data for adding {endpoint}:', table_data,
+                                      removeNull=True)
+    if testing:
+        return readable_output, outputs, result
+    return_outputs(readable_output, outputs, result)
+
+
+def format_update_object(result: dict, endpoint: str, testing: bool = False):
+    """
+        Formats updated object result from CheckPoint into Demisto's outputs.
+
+    Args:
+        result (dict): the report from CheckPoint.
+        endpoint (str): The endpoint that needs to be formatted.
+        testing(bool): is this function call is for testing purposes.
+
+    Returns:
+        str: the markdown to display inside Demisto.
+        dict: the context to return into Demisto.
+        dict: the report from CheckPoint (used for debugging).
+    """
+    outputs = {f'Checkpoint.{endpoint}': {
```

Comment:
add a DT to override the original object in context (see https://xsoar.pan.dev/docs/integrations/dt)
```suggestion
    outputs = {f'Checkpoint.{endpoint}(val.uid && val.uid == obj.uid)': {
```

this is relevant for all context objects returned in the integration.